pkg/sdk/testint/databases_integration_test.go

```diff
@@ -80,248 +94,712 @@ func TestInt_DatabasesCreate(t *testing.T) {
+				ToAccounts: []sdk.AccountIdentifier{
+					secondaryTestClientHelper().Ids.AccountIdentifierWithLocator(),
+				},
+				IgnoreEditionCheck: sdk.Bool(true),
+			},
+		})
+		require.NoError(t, err)
+
+		err = testClient(t).Databases.AlterReplication(ctx, database.ID(), &sdk.AlterDatabaseReplicationOptions{
+			DisableReplication: &sdk.DisableReplication{
+				ToAccounts: []sdk.AccountIdentifier{
+					secondaryTestClientHelper().Ids.AccountIdentifierWithLocator(),
+				},
+			},
 		})
 		require.NoError(t, err)
 	})
 
-	t.Run("setting and unsetting retention time + comment ", func(t *testing.T) {
-		databaseTest, _ := testClientHelper().Database.CreateDatabase(t)
-		err := client.Databases.Alter(ctx, databaseTest.ID(), &sdk.AlterDatabaseOptions{
-			Set: &sdk.DatabaseSet{
-				DataRetentionTimeInDays: sdk.Int(42),
-				Comment:                 sdk.String("test comment"),
+	t.Run("refresh replicated database", func(t *testing.T) {
+		client := testClient(t)
+		secondaryClient := testSecondaryClient(t)
+		ctx := testContext(t)
+
+		sharedDatabase, sharedDatabaseCleanup := secondaryTestClientHelper().Database.CreateDatabase(t)
+		t.Cleanup(sharedDatabaseCleanup)
+
+		err := secondaryClient.Databases.AlterReplication(ctx, sharedDatabase.ID(), &sdk.AlterDatabaseReplicationOptions{
+			EnableReplication: &sdk.EnableReplication{
+				ToAccounts: []sdk.AccountIdentifier{
+					testClientHelper().Account.GetAccountIdentifier(t),
+				},
+				IgnoreEditionCheck: sdk.Bool(true),
 			},
 		})
 		require.NoError(t, err)
-		database, err := client.Databases.ShowByID(ctx, databaseTest.ID())
+
+		externalVolume, externalVolumeCleanup := testClientHelper().ExternalVolume.Create(t)
+		t.Cleanup(externalVolumeCleanup)
+
+		catalog, catalogCleanup := testClientHelper().CatalogIntegration.Create(t)
+		t.Cleanup(catalogCleanup)
+
+		externalDatabaseId := sdk.NewExternalObjectIdentifier(secondaryTestClientHelper().Ids.AccountIdentifierWithLocator(), sharedDatabase.ID())
+		comment := random.Comment()
+		err = client.Databases.CreateSecondary(ctx, sharedDatabase.ID(), externalDatabaseId, &sdk.CreateSecondaryDatabaseOptions{
+			IfNotExists:                sdk.Bool(true),
+			DataRetentionTimeInDays:    sdk.Int(1),
+			MaxDataExtensionTimeInDays: sdk.Int(10),
+			ExternalVolume:             &externalVolume,
+			Catalog:                    &catalog,
+			DefaultDDLCollation:        sdk.String("en_US"),
+			LogLevel:                   sdk.Pointer(sdk.LogLevelDebug),
+			TraceLevel:                 sdk.Pointer(sdk.TraceLevelAlways),
+			Comment:                    sdk.String(comment),
+		})
 		require.NoError(t, err)
-		assert.Equal(t, 42, database.RetentionTime)
-		assert.Equal(t, "test comment", database.Comment)
-		err = client.Databases.Alter(ctx, databaseTest.ID(), &sdk.AlterDatabaseOptions{
-			Unset: &sdk.DatabaseUnset{
-				DataRetentionTimeInDays: sdk.Bool(true),
-				Comment:                 sdk.Bool(true),
+		t.Cleanup(func() {
```

Comment:
use testClientHelper().Database.DropDatabaseFunc(t, id), there is a problem with dropping databases, because it messes with the sessions context (especially the schema used).

This may be a bigger problem because there are multipledatabase drops in this file.

Reply:
Yeah, I somehow missed that one, changed.

Reply:
There are still a couple that need to be changed